s3_deploy.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+SRC_DIR='dist'
+DISTRIBUTION_ID='E1YPVV3YLYS4J7'
+# name of branch to deploy to root of site
+PRODUCTION_BRANCH='production'
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+	echo "skipping deploy to S3: this is a pull request"
+	exit 0
+fi
+
+# extract current TAG if present
+# the 2> is to prevent error messages when no match is found
+CURRENT_TAG=`git describe --tags --exact-match $TRAVIS_COMMIT 2> /dev/null`
+
+# strip PT ID from branch name for branch builds
+DEPLOY_DIR_NAME=$TRAVIS_BRANCH
+PT_PREFIX_REGEX="^([0-9]{8,}-)(.+)$"
+PT_SUFFIX_REGEX="^(.+)(-[0-9]{8,})$"
+if [[ $DEPLOY_DIR_NAME =~ $PT_PREFIX_REGEX ]]; then
+  DEPLOY_DIR_NAME=${BASH_REMATCH[2]}
+fi
+if [[ $DEPLOY_DIR_NAME =~ $PT_SUFFIX_REGEX ]]; then
+  DEPLOY_DIR_NAME=${BASH_REMATCH[1]}
+fi
+
+# tagged builds deploy to /version/TAG_NAME
+if [ "$TRAVIS_BRANCH" = "$CURRENT_TAG" ]; then
+  mkdir -p _site/version
+  DEPLOY_DIR="version/$TRAVIS_BRANCH"
+  DEPLOY_DEST= "_site/$DEPLOY_DIR"
+  INVAL_PATH="/version/$TRAVIS_BRANCH/index.html"
+  export DEPLOY_DIR
```

Comment:
Yeah it is used here: https://github.com/concord-consortium/collaborative-learning/pull/2/files#diff-16e2386ec454cf94fed2be96973ba154R19  it used so the s3 website code only looks at files in this subfolder on s3. Otherwise it requests hashes for every file from s3 in order to compare them with the local files.

Reply:
Yes, of course. Thanks for pointing that out -- you'd think I might have noticed that somewhere along the way given the number of times I've done this now. 😉